.github/workflows/firmware_build_test.yml

```diff
@@ -16,7 +16,11 @@ jobs:
         repository: PX4/Firmware
         ref: master
         submodules: recurvise
-    - run: git submodule update --init --recursive
+    - name: Checkout matching branch on PX4/Firmware if possible
```

Comment:
@kamilritz How is this related to this PR?

Reply:
This PR is breaking the interface to Firmware. If you provide a branch with the same name on upstream, it will use it in the Firmware Build tests. That is the reason why all the checks are passing.